canal.cc

```diff
@@ -271,16 +318,15 @@ mainExcept(int argc, char *argv[])
     }
     if (showsyms)
        exit(0);
-    sort(listed.begin() , listed.end() , compareSymbolsByAddress);
 
     // Now run through the corefile, searching for virtual objects.
     Elf::Off filesize = 0;
     Elf::Off memsize = 0;
 #ifdef WITH_PYTHON
     PythonPrinter<2> py(*process, std::cout, PstackOptions());
 #endif
+    std::vector<Elf::Off> data;
```

Comment:
I think you can reduce the scope of data to were it's used below, and construct it with the size as its known

Reply:
In conjunction with our memory capping that you recommended below, we could keep the vector here and reserve the max size (I though 64Ki) so we just allocate memory for the vector once.

Reply:
Sure, sounds good.